src/documents/ContextAwareness.md

```diff
@@ -0,0 +1,327 @@
+For lightweight cases of context adaptation, the [`distributeOptions`](IoCSS.md) scheme suffices, in that it enables a component to be responsive to broadcasts "advising" it from around the
+component tree in a free-form and controllable away. However, when a component needs to take a clearer role in responding to potentially many kinds of requests for adaptation, where these
+can be organised around two or more "axes" or "dimensions" of adaptation that can be clearly named and identified, it should opt into this more heavywight scheme by implementing the 
+`fluid.contextAware` grade and perhaps also advertising some of its dimensions of adaptations in its own options. These dimensions themselves can always be extended by further contributions
+to the component's options through all the usual sources - direct options, subcomponent options, and options distributions from elsewhere in the tree.
+
+A `contextAware` component can be responsive to contexts, whose influence is organised as a set of mutually orthogonal "adaptations", where the set of adaptations itself is open to extension by 
+integrators and adopters. The primary grade enabling a component to opt into this scheme is named `fluid.contextAware`. Any component derived from this grade will advertise an
+area of its options named `contextAwareness` which organises the rules for its adaptation in a hierarchical way - at the top level by "adaptation" and then at the nested level by
+"checks".
+
+## Structure of members in a `adaptationRecord`
+
+The top level structure of the `contextAwareness` record consists of a free hash of `adaptationName` strings to `adaptationRecord` structures. The `adaptationName` strings will be considered as
+namespaces for the purposes of priority resolution (see the <code>priority</code> entry in the following table). The elements of the `adaptationRecord` are
+described in the following table:
+
+<table>
+    <thead>
+        <tr>
+            <th colspan="3">Members of a <code>adaptationRecord</code> entry within the <code>contextAwareness</code> block of a <code>fluid.contextAware</code> component</th>
+        </tr>
+        <tr>
+            <th>Member</th>
+            <th>Type</th>
+            <th>Description</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td><code>checks</code> (optional)</td>
+            <td>Hash of <code>checkNamespace</code> to <code>checkRecord</code></td>
+            <td>A free hash of namespace names for checks, to entries describing how a context check is to be made, and what <code>gradeNames</code> should result if the check is successful.</td>
+        </tr>
+        <tr>
+            <td><code>defaultGradeNames</code> (optional)</td>
+            <td><code>String</code> or <code>Array of String</code></td>
+            <td>One or more <code>gradeNames</code> to be the result if none of the <code>checks</code> entries matches</td>
```

Comment:
I'm not sure how I could make this much clearer. I could talk about a possible process where one user had contributed a dimension where all they were sure about was the default grade in case no check matched, and further users then contributed checks which created conditions in which the default no longer always matched, but my feeling is that this would just represent a distraction to the user trying to learn from the docs. What do you think?